### PR TITLE
Updated API links in guides - phase 4

### PR DESCRIPTION
**Pull request by neojp**
This PR addresses #1440 for Ember `v2.17.0` and `v3.0.0`.

## Process

1. Ran the script `npm run release:guides:links <guides-path> <version>`
2. Manually checked updated links
3. Ran `npm run test:node-local` to check all external links automatically

### CODE REVIEW

**Comment by ijlee2** [comment]
I rebased the branch so that the pull request will be checked against GitHub Actions instead of Travis.

File: guides/v2.17.0/models/customizing-adapters.md [hunk 1]
```diff
@@ -235,7 +235,7 @@ export default DS.JSONAPIAdapter.extend({
 In some cases, your dynamic headers may require data from some
 object outside of Ember's observer system (for example
 `document.cookie`). You can use the
-[volatile](https://api.emberjs.com/ember/2.16/classes/@ember%2Fobject%2Fcomputed/methods/property?anchor=volatile&show=inherited%2Cprotected%2Cprivate%2Cdeprecated)
+[volatile](https://api.emberjs.com/ember/2.17/classes/@ember%2Fobject%2Fcomputed/methods/property?anchor=volatile&show=inherited%2Cprotected%2Cprivate%2Cdeprecated)
```
Comment (ijlee2) [inline guides/v2.17.0/models/customizing-adapters.md:238]:
```suggestion
[volatile](https://api.emberjs.com/ember/2.17/classes/@ember%2Fobject%2Fcomputed/methods/property?anchor=volatile)
```

File: guides/v2.17.0/models/customizing-serializers.md [hunk 2]
```diff
@@ -198,7 +198,7 @@ export default DS.JSONAPISerializer.extend({
 
 Similarly, if your backend store provides data in a format other than JSON API,
 you can use the
-[`normalizeResponse()`](https://api.emberjs.com/ember-data/2.16/classes/DS.JSONAPISerializer/methods/serialize?anchor=normalizeResponse)
+[`normalizeResponse()`](https://api.emberjs.com/ember-data/2.17/classes/DS.JSONAPISerializer/methods/serialize?anchor=normalizeResponse)
```
Comment (ijlee2) [inline guides/v2.17.0/models/customizing-serializers.md:201]:
```suggestion
[`normalizeResponse()`](https://api.emberjs.com/ember-data/2.17/classes/DS.JSONAPISerializer/methods/normalizeResponse?anchor=normalizeResponse)
```

File: guides/v2.17.0/models/customizing-serializers.md [hunk 3]
```diff
@@ -252,11 +252,11 @@ export default DS.JSONAPISerializer.extend({
 ```
 
 To normalize only a single model, you can use the
-[`normalize()`](https://api.emberjs.com/ember-data/2.16/classes/DS.JSONAPISerializer/methods/serialize?anchor=normalize)
+[`normalize()`](https://api.emberjs.com/ember-data/2.17/classes/DS.JSONAPISerializer/methods/serialize?anchor=normalize)
```
Comment (ijlee2) [inline guides/v2.17.0/models/customizing-serializers.md:255]:
```suggestion
[`normalize()`](https://api.emberjs.com/ember-data/2.17/classes/DS.JSONAPISerializer/methods/normalize?anchor=normalize)
```

File: guides/v2.17.0/models/customizing-serializers.md [hunk 4]
```diff
@@ -421,7 +421,7 @@ The JSON should encode the relationship as an ID to another record:
 ```
 If needed these naming conventions can be overwritten by implementing
 the
-[`keyForRelationship()`](https://api.emberjs.com/ember-data/2.16/classes/DS.JSONAPISerializer/methods/keyForAttribute?anchor=keyForRelationship)
+[`keyForRelationship()`](https://api.emberjs.com/ember-data/2.17/classes/DS.JSONAPISerializer/methods/keyForAttribute?anchor=keyForRelationship)
```
Comment (ijlee2) [inline guides/v2.17.0/models/customizing-serializers.md:424]:
```suggestion
[`keyForRelationship()`](https://api.emberjs.com/ember-data/2.17/classes/DS.JSONAPISerializer/methods/keyForRelationship?anchor=keyForRelationship)
```

File: guides/v2.17.0/models/finding-records.md [hunk 5]
```diff
@@ -2,14 +2,14 @@ The Ember Data store provides an interface for retrieving records of a single ty
...
-Use [`store.findRecord()`](https://api.emberjs.com/ember-data/2.16/classes/DS.Store/methods/findRecord?anchor=findRecord) to retrieve a record by its type and ID.
+Use [`store.findRecord()`](https://api.emberjs.com/ember-data/2.17/classes/DS.Store/methods/findRecord?anchor=findRecord) to retrieve a record by its type and ID.
 This will return a promise that fulfills with the requested record:
 
 ```javascript
 let blogPost = this.get('store').findRecord('blog-post', 1); // => GET /blog-posts/1
 ```
 
-Use [`store.peekRecord()`](https://api.emberjs.com/ember-data/2.16/classes/DS.Store/methods/findRecord?anchor=peekRecord) to retrieve a record by its type and ID, without making a network request.
+Use [`store.peekRecord()`](https://api.emberjs.com/ember-data/2.17/classes/DS.Store/methods/findRecord?anchor=peekRecord) to retrieve a record by its type and ID, without making a network request.
```
Comment (ijlee2) [inline guides/v2.17.0/models/finding-records.md:12]:
```suggestion
Use [`store.peekRecord()`](https://api.emberjs.com/ember-data/2.17/classes/DS.Store/methods/peekRecord?anchor=peekRecord) to retrieve a record by its type and ID, without making a network request.
```

File: guides/v2.17.0/models/finding-records.md [hunk 6]
```diff
@@ -59,7 +59,7 @@ this.get('store').query('person', {
 ### Querying for A Single Record
 
 If you are using an adapter that supports server requests capable of returning a single model object,
-Ember Data provides a convenience method [`store.queryRecord()`](https://api.emberjs.com/ember-data/2.16/classes/DS.Store/methods/query?anchor=queryRecord)that will return a promise that resolves with that single record.
+Ember Data provides a convenience method [`store.queryRecord()`](https://api.emberjs.com/ember-data/2.17/classes/DS.Store/methods/query?anchor=queryRecord)that will return a promise that resolves with that single record.
```
Comment (ijlee2) [inline guides/v2.17.0/models/finding-records.md:62]:
```suggestion
Ember Data provides a convenience method [`store.queryRecord()`](https://api.emberjs.com/ember-data/2.17/classes/DS.Store/methods/queryRecord?anchor=queryRecord) that will return a promise that resolves with that single record.
```

File: guides/v2.17.0/models/finding-records.md [hunk 7]
```diff
@@ -87,7 +87,7 @@ export default DS.Adapter.extend({
 });
 ```
 
-then calling [`store.queryRecord()`](https://api.emberjs.com/ember-data/2.16/classes/DS.Store/methods/query?anchor=queryRecord) will retrieve that object from the server:
+then calling [`store.queryRecord()`](https://api.emberjs.com/ember-data/2.17/classes/DS.Store/methods/query?anchor=queryRecord) will retrieve that object from the server:
```
Comment (ijlee2) [inline guides/v2.17.0/models/finding-records.md:90]:
```suggestion
then calling [`store.queryRecord()`](https://api.emberjs.com/ember-data/2.17/classes/DS.Store/methods/queryRecord?anchor=queryRecord) will retrieve that object from the server:
```

File: guides/v2.17.0/models/pushing-records-into-the-store.md [hunk 8]
```diff
@@ -78,7 +78,7 @@ the casing of the properties defined on the Model class.
 
 If you would like the data to be normalized by the model's default
 serializer before pushing it into the store, you can use the
-[`store.pushPayload()`](https://api.emberjs.com/ember-data/2.16/classes/DS.Store/methods/push?anchor=pushPayload) method.
+[`store.pushPayload()`](https://api.emberjs.com/ember-data/2.17/classes/DS.Store/methods/push?anchor=pushPayload) method.
```
Comment (ijlee2) [inline guides/v2.17.0/models/pushing-records-into-the-store.md:81]:
```suggestion
[`store.pushPayload()`](https://api.emberjs.com/ember-data/2.17/classes/DS.Store/methods/pushPayload?anchor=pushPayload) method.
```

**Review by ijlee2 — CHANGES_REQUESTED** [review]
I visited all links in the first 15 files. Before I check the rest, can you fix all the occurrences of `bar?anchor=foo` (i.e. `foo` and `bar` don't match) in 2.17 and 3.0?

If you can make 1 or 2 commits, instead of applying the suggestion (which would trigger the CI to run multiple times), that'd be good.

**Comment by jenweber** [comment]
@neojp do you have some time to apply these updates? We'd love to get this merged in. Thank you!

**Comment by neojp** [comment]
I will work on this during the weekend. Thanks for the reminder!

**Comment by locks** [comment]
Poke poke 😁

**Comment by neojp** [comment]
@ijlee2 I've merged your changes suggestions into a single commit. Sorry for the delay, it's been a long month!

**Comment by ijlee2** [comment]
@neojp No worries. I hope you are well and appreciate your taking time to work on this issue!

**Review by ijlee2 — APPROVED** [review]
🎉

**Pull request merged**